apollo-runtime/src/main/java/com/apollographql/apollo/subscription/AppSyncOperationMessageSerializer.kt

```diff
@@ -0,0 +1,86 @@
+package com.apollographql.apollo.subscription
+
+import com.apollographql.apollo.api.internal.json.JsonWriter
+import com.apollographql.apollo.api.internal.json.Utils
+import com.apollographql.apollo.api.internal.json.writeObject
+import com.apollographql.apollo.subscription.ApolloOperationMessageSerializer.writePayloadContentsTo
+import okhttp3.HttpUrl
+import okio.Buffer
+import okio.BufferedSink
+import okio.BufferedSource
+import java.net.URL
+import java.util.Base64
+
+class AppSyncOperationMessageSerializer(
+    private val authorization: Map<String, Any?>
+) : OperationMessageSerializer {
+  override fun OperationClientMessage.writeTo(sink: BufferedSink) {
+    when (this) {
+      is OperationClientMessage.Start -> JsonWriter.of(sink).use { writeTo(it) }
+      is OperationClientMessage.Init,
+      is OperationClientMessage.Stop,
+      is OperationClientMessage.Terminate -> with(ApolloOperationMessageSerializer) { writeTo(sink) }
```

Comment:
This reads weirdly. I know it works but it can be tempting to replace with `ApolloOperationMessageSerializer.writeTo(sink)`, which doesn't. That's an argument to rename `writeTo` to `writeClientMessage(sink, message)`. This is what moshi does for an exemple and will be less surprising for users?

Reply:
Yeah, the caller ergonomics isn't great. Will change.